python/dash_tools/mp4.py

```diff
@@ -837,20 +835,36 @@ def childpos(self):
         return self.offset+36
 
 
-class dac3_box(SampleEntry):
+class dac3_box(box):
     def __init__(self, *args):
-        SampleEntry.__init__(self, *args)
+        box.__init__(self, *args)
         self.dec_info = self.fmap[self.offset+8:self.offset+self.size]
         self.dec_info_hex = ''.join(['%02x' % ord(c) for c in self.dec_info])
         self.decoration = 'dec_info={0}'.format(self.dec_info_hex)
 
 
-class dec3_box(SampleEntry):
+class dec3_box(box):
     def __init__(self, *args):
-        SampleEntry.__init__(self, *args)
+        box.__init__(self, *args)
         self.dec_info = self.fmap[self.offset+8:self.offset+self.size]
         self.dec_info_hex = ''.join(['%02x' % ord(c) for c in self.dec_info])
-        self.decoration = 'dec_info={0}'.format(self.dec_info_hex)
+
+        # https://www.etsi.org/deliver/etsi_ts/102300_102399/102366/01.03.01_60/ts_102366v010301p.pdf
+        # EC3SpecificBox F.6.1 page 202
+        self.data_rate = (ord(self.dec_info[0]) << 5) + ((ord(self.dec_info[1]) >> 3) & 0x1f)
+        self.num_ind_sub = ord(self.dec_info[1]) & 0x7
+
+        # NOTE: currently only parsing the first dependent substream
```

Comment:
IMO, the comment is not correct. The first IN-dependent substream is parsed. Dependent sub streams, as indicated by num_dep_sub > 0 are not parsed.

Reply:
Yes I agree, I have pushed an update of the comment.